### PR TITLE
Upgrade to Rails 7

### PR DESCRIPTION
**Pull request by jcoyne**
## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test create_preassembly_image_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/create_preassembly_image_spec.rb) on stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

Ran `create_preassembly_image_spec.rb` on QA.

### CODE REVIEW

File: app/services/audit/checksum.rb [hunk 1]
```diff
@@ -34,6 +34,8 @@ def self.validate_list_of_druids(druid_list_file_path)
 
     # validate objects with a particular status on a particular moab_storage_root
     def self.validate_status_root(status, storage_root_name)
+      raise ArgumentError, "invalid status #{status}" unless CompleteMoab.statuses.key?(status)
```
Comment (justinlittman) [inline app/services/audit/checksum.rb:37]: AR used to object to querying on an invalid enum value. It no longer does, so explicitly checking and raising.

File: Gemfile [hunk 2]
```diff
@@ -1,5 +1,7 @@
 source 'https://rubygems.org'
 
+gem 'rails', '~> 7.0.0'
+
```
Comment (mjgiarlo) [inline Gemfile:4]: I have a slight preference for keeping the gems in alpha order but would not block over this. Unless you found that declaring this dependency above the others was required for some reason. I wouldn't be too surprised...

File: Gemfile.lock [hunk 3]
```diff
@@ -403,11 +407,10 @@ GEM
     websocket-extensions (0.1.5)
     whenever (1.0.0)
       chronic (>= 0.6.3)
-    zeitwerk (2.6.1)
+    zeitwerk (2.6.2)
 
 PLATFORMS
-  ruby
```
Comment (mjgiarlo) [inline Gemfile.lock:409]: I'm not sure if we need this platform or not. Do you recall? If not, what would you think about deploying this branch to QA and running an integration test to confirm we don't need the `ruby` platform in deployed envs?

File: app/services/audit/checksum.rb (hunk 1 above)
Comment (mjgiarlo) [inline app/services/audit/checksum.rb:37]: Is this part of the Rails 7 upgrade or extra goodness?

File: spec/services/audit/checksum_spec.rb [hunk 4]
```diff
@@ -91,8 +91,8 @@
       end
     end
 
-    it 'with invalid status, raises StatementInvalid' do
-      expect { described_class.validate_status_root('foo', root_name) }.to raise_error(ActiveRecord::StatementInvalid)
```
Comment (mjgiarlo) [inline spec/services/audit/checksum_spec.rb:95]: Ah, I'm guessing this behavior changed between Rails 6 and 7?

File: Gemfile (hunk 2 above)
Comment (justinlittman) [reply Gemfile:4]: That's where Jcoyne put it. Let me try moving and see what happens.

**Review by mjgiarlo — APPROVED** [review]
I have a few minor questions and comments. Feel free to reply & edit or just merge. 😄

File: Gemfile.lock (hunk 3 above)
Comment (justinlittman) [reply Gemfile.lock:409]: Oh wait, I did and it worked.

Comment (mjgiarlo) [reply Gemfile.lock:409]: OK, great. 🎉

File: Gemfile (hunk 2 above)
Comment (mjgiarlo) [reply Gemfile:4]: kk, appreciate it

File: app/services/audit/checksum.rb (hunk 1 above)
Comment (mjgiarlo) [reply app/services/audit/checksum.rb:37]: 🥇

Comment (justinlittman) [reply app/services/audit/checksum.rb:37]: Hmmm. I thought I put a comment about this.

Rails used to validate the values of enums during a query. It no longer does.

There was a test for this, so above change replicates the enum check. I doubt it is crucial, but there was a test.

Comment (mjgiarlo) [reply app/services/audit/checksum.rb:37]: 🚀

**Pull request merged**